builtin/providers/scaleway/helpers.go

```diff
@@ -17,36 +19,55 @@ func String(val string) *string {
 	return &val
 }
 
-// NOTE copied from github.com/scaleway/scaleway-cli/pkg/api/helpers.go
-// the helpers.go file pulls in quite a lot dependencies, and they're just convenience wrappers anyway
+// deleteRunningServer terminates the server and waits until it is removed.
+func deleteRunningServer(scaleway *api.ScalewayAPI, server *api.ScalewayServer) error {
+	err := scaleway.PostServerAction(server.Identifier, "terminate")
 
-func deleteServerSafe(s *api.ScalewayAPI, serverID string) error {
-	server, err := s.GetServer(serverID)
 	if err != nil {
+		if serr, ok := err.(api.ScalewayAPIError); ok {
+			if serr.StatusCode == 404 {
+				return nil
+			}
+		}
+
 		return err
 	}
 
-	if server.State != "stopped" {
-		if err := s.PostServerAction(serverID, "poweroff"); err != nil {
-			return err
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
```

Comment:
5 minutes seams very optimistic considering the current performances of scaleway, any way to make this with config at the provider level and/or to put 10 minutes ?